doc/developers.md

```diff
@@ -330,12 +332,20 @@ $ pytest ../tests/system/
 ```
 The [`test_system.py`](https://github.com/UCL/TDMS/blob/main/tdms/tests/system/test_system.py) script runs each system test in sequence.
 
-**[`test_regen.py`](https://github.com/UCL/TDMS/blob/main/tdms/tests/system/test_regen.py) and [`tdms_testing_class.py`](https://github.com/UCL/TDMS/blob/main/tdms/tests/system/tdms_testing_class.py.py) will replace `test_system.py` and `read_config.py` when the [input overhaul](https://github.com/UCL/TDMS/issues/70) is complete.**
-
-When you run the tests for the first time, test data is downloaded to `tdms/tests/system/data` (and will be [ignored by git](https://github.com/UCL/TDMS/blob/main/.gitignore)).
-These reference input files contain arrays for: the incident electric field, the computational grid, etc. which are needed by the simulation, and have been generated by a trusted version of the relevant MATLAB scripts.
+When you run the tests for the first time, the test data is downloaded to `tdms/tests/system/data` (and will be [ignored by git](https://github.com/UCL/TDMS/blob/main/.gitignore)).
 Subsequent runs of the tests will not re-download unless you manually delete the zip file(s).
 
+The test data contains "reference" input files which in turn provide arrays for: the incident electric field, the computational grid, etc. which are needed by the simulation, and have been generated by a trusted version of the relevant MATLAB scripts.
+The data also contains "reference" output files which have been generated by a trusted version of TDMS - the system tests check that any development changes have preserved the functionality of TDMS.
+
+<details>
+<summary> **Planned changes regarding MATLAB removal** </summary>
+\note [`test_regen.py`](https://github.com/UCL/TDMS/blob/main/tdms/tests/system/test_regen.py) will replace `test_system.py` and `read_config.py` when the [input overhaul](https://github.com/UCL/TDMS/issues/70) is complete. Due to difficulties running `MATLAB` from within `Python` on CI (see [#239](https://github.com/UCL/TDMS/issues/239)), we can only run `test_system.py` on CI, which is why the test data contains input `.mat` files as well as the corresponding outputs.
+
+If you are developing, you can run `test_regen.py` yourself in place of `test_system.py` by invoking `pytest`. However you will need to [install `MatlabEngine`](https://uk.mathworks.com/help/matlab/matlab_external/install-the-matlab-engine-for-python.html) into the environment you installed the `requirements.txt` first.
+</details>
+
```

Comment:
```suggestion
The test data contains reference input files which contain arrays for the incident electric field, the computational grid, and so on (needed by the simulation).
The files have been generated by a trusted version of the relevant MATLAB scripts.
The data also contains reference _output_ files which have been generated by a trusted version of TDMS - the system tests check output against reference outputs.
I.e. regression testing: that new changes have preserved the functionality of TDMS.
```

Can we delete the details? We either will or won't remove MATLAB then update the docs accordingly.

In case you want to keep the details, I'd explicitly mention "regression tests" in the description. Because you use a fair few words to describe regression tests (which has a wikipedia page). But don't say the name.